File: solutions/clojure/space-age/src/space_age.clj
(ns space-age)

(def ^:const seconds-in-earth-year 31557600)

(defn- get-local-age [ratio earth-seconds]
  (/ earth-seconds ratio seconds-in-earth-year))

(def on-mercury 
  "Local age on Mercury represented by passed seconds on Earth"
  (partial get-local-age 0.2408467))

(def on-venus
  "Local age on Venus represented by passed seconds on Earth"
  (partial get-local-age 0.61519726))

(def on-earth
  "Local age on Earth represented by passed seconds on Earth"
  (partial get-local-age 1))

(def on-mars
  "Local age on Mars represented by passed seconds on Earth"
  (partial get-local-age 1.8808158))  

(def on-jupiter
  "Local age on Jupiter represented by passed seconds on Earth"
  (partial get-local-age 11.862615))

(def on-saturn
  "Local age on Saturn represented by passed seconds on Earth"
  (partial get-local-age 29.447498))

(def on-neptune
  "Local age on Neptune represented by passed seconds on Earth"
  (partial get-local-age 164.79132))  

(def on-uranus
  "Local age on Uranus represented by passed seconds on Earth"
  (partial get-local-age 84.016846))

File: solutions/typescript/space-age/space-age.ts
const EARTH_ORBITAL_LENGTH = 31_557_600;

type Planet = keyof typeof OrbitalRatios

enum OrbitalRatios {
  mercury = 0.2408467,
  venus = 0.61519726,
  earth = 1.0,
  mars = 1.8808158,
  jupiter = 11.862615,
  saturn = 29.447498,
  uranus = 84.016846,
  neptune = 164.79132,
}

export function age(planet: Planet, seconds: number): number {
  const localOrbitalLength = OrbitalRatios[planet] * EARTH_ORBITAL_LENGTH
  const localAge = seconds / localOrbitalLength
  return Number(localAge.toFixed(2));
}